src/main/java/frc/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,284 @@
+
+import static edu.wpi.first.units.Units.*;
+
+import edu.wpi.first.units.measure.Angle;
+import edu.wpi.first.units.measure.Voltage;
+import edu.wpi.first.wpilibj.Alert;
+import edu.wpi.first.wpilibj.Alert.AlertType;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SelectCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import java.util.Map;
+import org.littletonrobotics.junction.AutoLogOutput;
+import org.littletonrobotics.junction.Logger;
+
+/**
+ * The Arm subsystem controls a dual-motor arm mechanism for game piece manipulation. It supports
+ * multiple positions for different game actions and provides both open-loop and closed-loop control
+ * options.
+ */
+public class Arm extends SubsystemBase {
+  // Hardware interface and inputs
+  private final ArmIO io;
+  private final ArmIOInputsAutoLogged inputs;
+
+  // Alerts for motor connection status
+  private final Alert leaderMotorAlert =
+      new Alert("Arm leader motor isn't connected", AlertType.kError);
+  private final Alert followerMotorAlert =
+      new Alert("Arm follower motor isn't connected", AlertType.kError);
+  private final Alert encoderAlert = new Alert("Arm encoder isn't connected", AlertType.kError);
+
+  // System identification routine configuration
+  private final SysIdRoutine sysId =
+      new SysIdRoutine(
+          new SysIdRoutine.Config(
+              null, // Use default ramp rate (1 V/s)
+              Volts.of(4), // Reduce dynamic step voltage to 4 to prevent brownout
+              null, // Use default timeout (10 s)
+              // Log state with Phoenix SignalLogger class
+              state -> Logger.recordOutput("state", state.toString())),
+          new SysIdRoutine.Mechanism(
+              output -> {
+                this.runVoltage(output);
+                Logger.recordOutput("Arm_Position", output.in(Volts));
+              },
+              null,
+              this));
+
+  // Current arm position mode
+  private ArmPosition currentMode = ArmPosition.STOWED;
+
+  /**
+   * Creates a new Arm subsystem with the specified hardware interface.
+   *
+   * @param io The hardware interface implementation for the arm
+   */
+  public Arm(ArmIO io) {
+    this.io = io;
+    this.inputs = new ArmIOInputsAutoLogged();
+  }
+
+  @Override
+  public void periodic() {
+    // Update and log inputs from hardware
+    io.updateInputs(inputs);
+    Logger.processInputs("Arm", inputs);
+
+    // Update motor connection status alerts
+    leaderMotorAlert.set(!inputs.leaderConnected);
+    followerMotorAlert.set(!inputs.followerConnected);
+    encoderAlert.set(!inputs.encoderConnected);
+  }
+
+  /**
+   * Runs the arm in open-loop mode at the specified voltage.
+   *
+   * @param volts The voltage to apply to the motors
+   */
+  public void runVoltage(Voltage volts) {
```

Comment:
same thing, maybe dont' want to expose this if you can force consumers to use the commands